File: src/lib/email.ts
import nodemailer from "nodemailer";

const transporter = nodemailer.createTransport({
  host: process.env.SMTP_HOST || "smtp.gmail.com",
  port: Number(process.env.SMTP_PORT) || 587,
  secure: false, // STARTTLS on port 587
  auth: {
    user: process.env.SMTP_USER,
    pass: process.env.SMTP_PASS,
  },
});

const defaultFrom =
  process.env.SMTP_FROM || "noreply@efilo.ai";

interface SendEmailParams {
  to: string;
  toName?: string;
  from?: string;
  fromName?: string;
  replyTo?: string;
  subject: string;
  html: string;
  text?: string;
}

/**
 * Low-level email sender via SMTP (Google Workspace).
 */
export async function sendEmail(params: SendEmailParams) {
  if (!process.env.SMTP_USER || !process.env.SMTP_PASS) {
    console.warn("[email] SMTP_USER/SMTP_PASS not set — skipping email send");
    return;
  }

  const from = params.fromName
    ? `"${params.fromName}" <${params.from || defaultFrom}>`
    : params.from || defaultFrom;

  const to = params.toName
    ? `"${params.toName}" <${params.to}>`
    : params.to;

  await transporter.sendMail({
    from,
    to,
    replyTo: params.replyTo,
    subject: params.subject,
    html: params.html,
    text: params.text,
  });
}

interface SendRfiEmailParams {
  fromName: string;
  fromEmail: string;
  replyTo: string;
  to: string;
  toName?: string;
  rfiNumber: string;
  subject: string;
  question: string;
  projectName: string;
}

/**
 * Send an RFI email to the GC contact.
 */
export async function sendRfiEmail(params: SendRfiEmailParams) {
  const html = `
<!DOCTYPE html>
<html>
<head><meta charset="utf-8"></head>
<body style="font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif; color: #1C1917; max-width: 600px; margin: 0 auto; padding: 20px;">
  <div style="border-bottom: 3px solid #C67F17; padding-bottom: 16px; margin-bottom: 24px;">
    <h2 style="margin: 0; color: #C67F17;">RFI ${params.rfiNumber}</h2>
    <p style="margin: 4px 0 0; color: #57534E; font-size: 14px;">Project: ${escapeHtml(params.projectName)}</p>
  </div>

  <p style="margin: 0 0 8px; font-weight: 600; font-size: 15px;">${escapeHtml(params.subject)}</p>

  <div style="background: #FAFAF8; border: 1px solid #E8E5DE; border-radius: 6px; padding: 16px; margin: 16px 0; white-space: pre-wrap; font-size: 14px; line-height: 1.6;">
${escapeHtml(params.question)}
  </div>

  <p style="color: #57534E; font-size: 13px; margin-top: 24px;">
    Sent by ${escapeHtml(params.fromName)}
  </p>
  <p style="color: #57534E; font-size: 12px; margin-top: 8px;">
    Please reply directly to this email to respond.
  </p>
</body>
</html>`.trim();

  const text = `RFI ${params.rfiNumber} — ${params.subject}\nProject: ${params.projectName}\n\n${params.question}\n\nSent by ${params.fromName}\nPlease reply directly to this email to respond.`;

  await sendEmail({
    to: params.to,
    toName: params.toName,
    from: params.fromEmail,
    fromName: params.fromName,
    replyTo: params.replyTo,
    subject: `RFI ${params.rfiNumber}: ${params.subject}`,
    html,
    text,
  });
}

function escapeHtml(str: string): string {
  return str
    .replace(/&/g, "&amp;")
    .replace(/</g, "&lt;")
    .replace(/>/g, "&gt;")
    .replace(/"/g, "&quot;");
}

File: backend/app/services/email.py
"""Email service (SMTP).

Sends RFI emails and other transactional emails via SMTP (Gmail/Google Workspace).
"""

import logging
import smtplib
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText

from app.config import get_settings

logger = logging.getLogger(__name__)

SMTP_HOST = "smtp.gmail.com"
SMTP_PORT = 587


def send_rfi_email(
    *,
    from_name: str,
    from_email: str,
    reply_to: str,
    to: str,
    to_name: str | None = None,
    cc: str | None = None,
    rfi_number: str,
    subject: str,
    question: str,
    project_name: str,
) -> bool:
    """Send an RFI email to the GC contact.

    Returns True on success, False on failure.
    """
    settings = get_settings()
    if not settings.smtp_user or not settings.smtp_pass:
        logger.warning("SMTP credentials not configured — skipping email send")
        return False

    email_subject = f"RFI {rfi_number}: {subject}"

    # Build HTML body
    html = f"""\
<html>
<body style="font-family: 'DM Sans', Arial, sans-serif; color: #1C1917; line-height: 1.6;">
  <div style="max-width: 600px; margin: 0 auto;">
    <h2 style="color: #C67F17; margin-bottom: 4px;">RFI {rfi_number}</h2>
    <p style="color: #57534E; margin-top: 0;">Project: {project_name}</p>
    <p style="font-weight: 600;">{subject}</p>
    <div style="background: #FAFAF8; border: 1px solid #E8E5DE; border-radius: 8px; padding: 16px; margin: 16px 0; white-space: pre-wrap;">
{question}
    </div>
    <p style="color: #57534E; font-size: 14px;">Sent by {from_name}</p>
    <p style="color: #57534E; font-size: 14px;">Please reply directly to this email to respond.</p>
    <hr style="border: none; border-top: 1px solid #E8E5DE; margin: 24px 0;" />
    <p style="color: #9CA3AF; font-size: 12px;">Sent via efilo.ai</p>
  </div>
</body>
</html>"""

    # Plain text fallback
    text = (
        f"RFI {rfi_number}\n"
        f"Project: {project_name}\n\n"
        f"{subject}\n\n"
        f"{question}\n\n"
        f"Sent by {from_name}\n"
        f"Please reply directly to this email to respond.\n"
    )

    msg = MIMEMultipart("alternative")
    msg["Subject"] = email_subject
    msg["From"] = f"{from_name} <{from_email}>"
    msg["To"] = f"{to_name} <{to}>" if to_name else to
    msg["Reply-To"] = reply_to
    if cc:
        msg["Cc"] = cc

    msg.attach(MIMEText(text, "plain"))
    msg.attach(MIMEText(html, "html"))

    recipients = [to]
    if cc:
        recipients.append(cc)

    try:
        with smtplib.SMTP(SMTP_HOST, SMTP_PORT) as server:
            server.ehlo()
            server.starttls()
            server.ehlo()
            server.login(settings.smtp_user, settings.smtp_pass)
            server.sendmail(from_email, recipients, msg.as_string())
        logger.info("RFI email sent: %s → %s", rfi_number, to)
        return True
    except Exception:
        logger.exception("Failed to send RFI email %s to %s", rfi_number, to)
        return False
